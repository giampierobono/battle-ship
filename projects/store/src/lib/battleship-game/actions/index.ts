export * from './battle-ship-game.actions';
