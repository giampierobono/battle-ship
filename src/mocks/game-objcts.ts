import { BoatModel } from '../models';

export const mockGameSettings = {
  boardSize: 8,
  boatsPerPlayer: 4,
  numOfPlayers: 2,
};

export const mockBoatShapes = {
  '0': {
    shapeType: 0,
    shapeIndex: 0,
    segments: [
      {
        rowsCount: 2,
        colsCount: 1,
      },
      {
        rowsCount: 1,
        colsCount: 2,
      },
    ],
    shapeSize: 4,
  },
  '1': {
    shapeType: 1,
    shapeIndex: 0,
    segments: [
      {
        rowsCount: 2,
        colsCount: 2,
      },
    ],
    shapeSize: 4,
  },
  '2': {
    shapeType: 2,
    shapeIndex: 0,
    segments: [
      {
        rowsCount: 1,
        colsCount: 4,
      },
    ],
    shapeSize: 4,
  },
};

export const mockAllBoats: Array<Array<BoatModel>> = [
  [
    {
      positions: {
        '2': {
          '5': 'BOAT',
        },
        '3': {
          '5': 'HIT',
        },
        '4': {
          '5': 'BOAT',
          '6': 'BOAT',
        },
      },
      playerIndex: 0,
      boatIndex: 0,
      shapeKey: 0,
      hits: 1,
    },
    {
      positions: {
        '0': {
          '6': 'BOAT',
        },
        '1': {
          '6': 'BOAT',
        },
        '2': {
          '6': 'BOAT',
          '7': 'BOAT',
        },
      } as any,
      playerIndex: 0,
      boatIndex: 1,
      shapeKey: 0,
      hits: 0,
    },
    {
      positions: {
        '0': {
          '1': 'BOAT',
          '2': 'BOAT',
        },
        '1': {
          '1': 'BOAT',
          '2': 'BOAT',
        },
      } as any,
      playerIndex: 0,
      boatIndex: 2,
      shapeKey: 1,
      hits: 0,
    },
    {
      positions: {
        '4': {
          '0': 'BOAT',
          '1': 'BOAT',
        },
        '5': {
          '0': 'BOAT',
          '1': 'BOAT',
        },
      } as any,
      playerIndex: 0,
      boatIndex: 3,
      shapeKey: 1,
      hits: 0,
    },
  ],
  [
    {
      positions: {
        '6': {
          '2': 'BOAT',
          '3': 'BOAT',
          '4': 'BOAT',
          '5': 'BOAT',
        },
      } as any,
      playerIndex: 1,
      boatIndex: 0,
      shapeKey: 2,
      hits: 0,
    },
    {
      positions: {
        '2': {
          '0': 'BOAT',
          '1': 'BOAT',
          '2': 'BOAT',
          '3': 'BOAT',
        },
      } as any,
      playerIndex: 1,
      boatIndex: 1,
      shapeKey: 2,
      hits: 0,
    },
    {
      positions: {
        '0': {
          '2': 'BOAT',
          '3': 'BOAT',
          '4': 'BOAT',
          '5': 'BOAT',
        },
      } as any,
      playerIndex: 1,
      boatIndex: 2,
      shapeKey: 2,
      hits: 0,
    },
    {
      positions: {
        '3': {
          '6': 'HIT',
        },
        '4': {
          '6': 'BOAT',
        },
        '5': {
          '6': 'BOAT',
          '7': 'BOAT',
        },
      } as any,
      playerIndex: 1,
      boatIndex: 3,
      shapeKey: 0,
      hits: 1,
    },
  ],
];
