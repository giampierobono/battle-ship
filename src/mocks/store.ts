export const mockState = {
  battleShipCore: {
    players: {
      players: [
        {
          playerIndex: 0,
        },
        {
          playerIndex: 1,
        },
      ],
    },
    boats: {
      boats: [
        [
          {
            positions: {
              '2': {
                '5': 'BOAT',
              },
              '3': {
                '5': 'HIT',
              },
              '4': {
                '5': 'BOAT',
                '6': 'BOAT',
              },
            },
            playerIndex: 0,
            boatIndex: 0,
            shapeKey: 0,
            hits: 1,
          },
          {
            positions: {
              '0': {
                '6': 'HIT',
              },
              '1': {
                '6': 'BOAT',
              },
              '2': {
                '6': 'BOAT',
                '7': 'HIT',
              },
            },
            playerIndex: 0,
            boatIndex: 1,
            shapeKey: 0,
            hits: 2,
          },
          {
            positions: {
              '0': {
                '1': 'BOAT',
                '2': 'BOAT',
              },
              '1': {
                '1': 'BOAT',
                '2': 'BOAT',
              },
            },
            playerIndex: 0,
            boatIndex: 2,
            shapeKey: 1,
            hits: 0,
          },
          {
            positions: {
              '4': {
                '0': 'BOAT',
                '1': 'BOAT',
              },
              '5': {
                '0': 'BOAT',
                '1': 'BOAT',
              },
            },
            playerIndex: 0,
            boatIndex: 3,
            shapeKey: 1,
            hits: 0,
          },
        ],
        [
          {
            positions: {
              '6': {
                '2': 'BOAT',
                '3': 'BOAT',
                '4': 'BOAT',
                '5': 'BOAT',
              },
            },
            playerIndex: 1,
            boatIndex: 0,
            shapeKey: 2,
            hits: 0,
          },
          {
            positions: {
              '2': {
                '0': 'BOAT',
                '1': 'BOAT',
                '2': 'BOAT',
                '3': 'BOAT',
              },
            },
            playerIndex: 1,
            boatIndex: 1,
            shapeKey: 2,
            hits: 0,
          },
          {
            positions: {
              '0': {
                '2': 'BOAT',
                '3': 'BOAT',
                '4': 'BOAT',
                '5': 'BOAT',
              },
            },
            playerIndex: 1,
            boatIndex: 2,
            shapeKey: 2,
            hits: 0,
          },
          {
            positions: {
              '3': {
                '6': 'HIT',
              },
              '4': {
                '6': 'BOAT',
              },
              '5': {
                '6': 'BOAT',
                '7': 'BOAT',
              },
            },
            playerIndex: 1,
            boatIndex: 3,
            shapeKey: 0,
            hits: 1,
          },
        ],
      ],
    },
    boatShapes: {
      shapes: {
        '0': {
          shapeType: 0,
          shapeIndex: 0,
          segments: [
            {
              rowsCount: 2,
              colsCount: 1,
            },
            {
              rowsCount: 1,
              colsCount: 2,
            },
          ],
          shapeSize: 4,
        },
        '1': {
          shapeType: 1,
          shapeIndex: 0,
          segments: [
            {
              rowsCount: 2,
              colsCount: 2,
            },
          ],
          shapeSize: 4,
        },
        '2': {
          shapeType: 2,
          shapeIndex: 0,
          segments: [
            {
              rowsCount: 1,
              colsCount: 4,
            },
          ],
          shapeSize: 4,
        },
      },
    },
    settings: {
      boardSize: 8,
      boatsPerPlayer: 4,
      numOfPlayers: 2,
    },
    game: {
      playerTurn: 1,
      isCurrentTurnOver: true,
      isGameOver: false,
      playersMoves: {
        '0': {
          '1': {
            '7': 'MISS',
          },
          '2': {
            '5': 'MISS',
          },
          '3': {
            '3': 'MISS',
            '6': 'HIT',
          },
          '5': {
            '5': 'MISS',
          },
        },
        '1': {
          '0': {
            '6': 'HIT',
          },
          '2': {
            '7': 'HIT',
          },
          '3': {
            '5': 'HIT',
          },
          '5': {
            '4': 'MISS',
            '6': 'MISS',
          },
        },
      },
      playersSunkBoats: {},
    },
  },
};
