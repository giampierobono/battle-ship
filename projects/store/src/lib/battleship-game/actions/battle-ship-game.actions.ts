import { createAction, props } from '@ngrx/store';
import { PlayerFiresPositionPayloadModel } from '../../players';

export const tryFirePosition = createAction('[Game] Player fire!', props<PlayerFiresPositionPayloadModel>());
