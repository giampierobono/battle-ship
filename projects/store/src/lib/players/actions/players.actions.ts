import { createAction, props } from '@ngrx/store';
import { AddBoatToPlayerPayload, CreatePlayerPayloadModel, HitPlayerBoatPayloadModel } from '../models';

export const createPlayer = createAction('[Players] Add new player', props<CreatePlayerPayloadModel>());

export const addBoatToPlayer = createAction('[Players] Add boat to player', props<AddBoatToPlayerPayload>());

export const hitPlayerBoat = createAction('[Player] Hit player boat', props<HitPlayerBoatPayloadModel>());
