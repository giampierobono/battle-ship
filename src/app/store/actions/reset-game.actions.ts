import { createAction } from '@ngrx/store';

export const resetGame = createAction('[BS webapp] Reset game');
