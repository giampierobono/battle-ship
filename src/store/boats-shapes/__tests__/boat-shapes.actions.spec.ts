import { addBoatShapes } from '../boat-shapes.actions';
import { mockBoatShapesArray } from '../../../mocks/game-objcts';

describe('Boat shapes actions', () => {
  describe('addBoatShapes', () => {
    it('should return correct action structure', () => {
      expect(addBoatShapes({ shapes: mockBoatShapesArray })).toMatchSnapshot();
    });
  });
});
