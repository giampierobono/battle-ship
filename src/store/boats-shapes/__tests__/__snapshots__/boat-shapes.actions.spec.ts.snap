// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boat shapes actions addBoatShapes should return correct action structure 1`] = `
Object {
  "shapes": Array [
    Object {
      "segments": Array [
        Object {
          "colsCount": 4,
          "rowsCount": 1,
        },
      ],
      "shapeIndex": 0,
      "shapeSize": 4,
      "shapeType": 2,
    },
    Object {
      "segments": Array [
        Object {
          "colsCount": 2,
          "rowsCount": 2,
        },
      ],
      "shapeIndex": 0,
      "shapeSize": 4,
      "shapeType": 1,
    },
    Object {
      "segments": Array [
        Object {
          "colsCount": 1,
          "rowsCount": 2,
        },
        Object {
          "colsCount": 2,
          "rowsCount": 1,
        },
      ],
      "shapeIndex": 0,
      "shapeSize": 4,
      "shapeType": 0,
    },
  ],
  "type": "[Boats] Add boat shape",
}
`;
